### PR TITLE
DDF-3190 Added QueryFilterTransformer Interface

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
This PR adds a QueryFilterTransformer interface.
#### Who is reviewing it? 
@troymohl 
@mackncheesiest 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
@codice/core-apis 
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@kcwire 
@millerw8
#### How should this be tested? (List steps with links to updated documentation)
A full build should suffice
#### Any background context you want to provide?
https://codice.atlassian.net/wiki/display/DDF/QueryFilterTransformer+Design
#### What are the relevant tickets?
[DDF-3190](https://codice.atlassian.net/browse/DDF-3190)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
build now

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/QueryFilterTransformer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Transforms a {@link QueryRequest} containing external attribute names & values into a
+     * normalized taxonomy {@link QueryRequest}. {@link QueryRequest} is used in order to allow
+     * the transformer to directly modify other properties (e.g. sourceId) based on criteria
+     * without requiring the endpoint to modify the request after conversion.  Transform method
+     * will use the FilterVisitor pattern and build the result Filter as the query is parsed.
+     *
+     * @return transformed {@link QueryRequest}
+     */
+    QueryRequest transform(QueryRequest queryRequest);
```
Comment (kcwire) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/QueryFilterTransformer.java]: In several other `Transformer` interfaces we include a properties map to allow additional information to be provided that may not be present in the object itself. I would strongly consider adding it to match the others.

**Comment by bdeining** [comment]
build now

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/QueryFilterTransformer.java (hunk 1 above)
Comment (bdeining) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/QueryFilterTransformer.java]: added a properties map and cleaned up the javadoc to be less assuming

**Pull request merged**